lib/css_parser/rule_set.rb

```diff
@@ -22,90 +24,152 @@ class RuleSet
       ['border-width', %w[border-top-width border-right-width border-bottom-width border-left-width]]
     ].freeze
 
-    class DeclarationValue
-      attr_reader :value
-      attr_accessor :important, :order
+    class Declarations
+      class Value
+        attr_reader :value
+        attr_accessor :important, :order
+
+        def initialize(value, important: nil, order: 0)
+          self.value = value
+          @important = important.nil? ? !value.match(CssParser::IMPORTANT_IN_PROPERTY_RX).nil? : important
+          @order = order
+        end
+
+        def value=(value)
+          value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
+          raise ArgumentError, 'value is empty' if value.empty?
+
+          @value = value.freeze
+        end
+
+        def to_s
+          return value unless important
 
-      def initialize(value, important: nil, order: 0)
-        self.value = value
-        @important = important.nil? ? !value.match(CssParser::IMPORTANT_IN_PROPERTY_RX).nil? : important
-        @order = order
+          "#{value} !important"
+        end
       end
 
-      def value=(value)
-        value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
-        raise ArgumentError, 'value is empty' if value.empty?
+      attr_reader :declarations
+      private :declarations
```

Comment:
could move that into private section

Reply:
... would actually prefer o use `@declations` since it's private

Reply:
I don't like accessing instance variable directly because typos becomes nils.